From: Luca Weiss <luca.weiss@fairphone.com>
Date: Thu, 17 Aug 2023 10:22:35 +0200
Subject: WIP arm64: dts: qcom: qcm6490-fairphone-fp5: Configure PM8008
 regulators

PM8008 regulators are used for the cameras found on FP5. Configure the
chip and its voltages.

Signed-off-by: Luca Weiss <luca.weiss@fairphone.com>

[WIP] Because pm8008 regulator bindings are still changing on mailing
          list, wait until upstream and then adjust.
---
 arch/arm64/boot/dts/qcom/qcm6490-fairphone-fp5.dts | 72 +++++++++++++++++++++-
 1 file changed, 71 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/qcm6490-fairphone-fp5.dts b/arch/arm64/boot/dts/qcom/qcm6490-fairphone-fp5.dts
index e7e20f7..0ad0032 100644
--- a/arch/arm64/boot/dts/qcom/qcm6490-fairphone-fp5.dts
+++ b/arch/arm64/boot/dts/qcom/qcm6490-fairphone-fp5.dts
@@ -393,7 +393,8 @@ &gpi_dma1 {
 &i2c1 {
 	status = "okay";
 
-	/* PM8008 PMIC @ 8 and 9 */
+	#include "pm8008.dtsi"
+
 	/* Pixelworks @ 26 */
 	/* FSA4480 USB audio switch @ 42 */
 	/* AW86927FCR haptics @ 5a */
@@ -436,6 +437,66 @@ volume_down_default: volume-down-default-state {
 	};
 };
 
+&pm8008_chip {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pm8008_active>;
+};
+
+&pm8008_regulators {
+	vdd_l1_l2-supply = <&vreg_s8b>;
+	vdd_l3_l4-supply = <&vreg_bob>;
+	vdd_l5-supply = <&vreg_bob>;
+	vdd_l6-supply = <&vreg_s1b>;
+	vdd_l7-supply = <&vreg_bob>;
+};
+
+&pm8008_l1 {
+	regulator-min-microvolt = <1000000>;
+	regulator-max-microvolt = <1200000>;
+	regulator-min-dropout-voltage-microvolt = <160000>;
+};
+
+&pm8008_l2 {
+	regulator-min-microvolt = <950000>;
+	/* Adjusted from downstream 1150000 to avoid regulator core failure
+	   since current_uV on boot is 1.2V, but the regulator steps don't
+	   include exactly 1.15V, just 1.152V */
+	regulator-max-microvolt = <1152000>;
+	regulator-min-dropout-voltage-microvolt = <128000>;
+};
+
+&pm8008_l3 {
+	regulator-min-microvolt = <2700000>;
+	regulator-max-microvolt = <3000000>;
+	regulator-min-dropout-voltage-microvolt = <144000>;
+};
+
+&pm8008_l4 {
+	regulator-min-microvolt = <2700000>;
+	regulator-max-microvolt = <2900000>;
+	regulator-min-dropout-voltage-microvolt = <128000>;
+};
+
+&pm8008_l5 {
+	/* Downstream min voltage: 2.7V. See comment at pm8008_l2 */
+	regulator-min-microvolt = <2704000>;
+	regulator-max-microvolt = <2900000>;
+	regulator-min-dropout-voltage-microvolt = <128000>;
+};
+
+&pm8008_l6 {
+	regulator-min-microvolt = <1700000>;
+	/* Downstream max voltage: 1.9V. See comment at pm8008_l2 */
+	regulator-max-microvolt = <1904000>;
+	regulator-min-dropout-voltage-microvolt = <136000>;
+};
+
+&pm8008_l7 {
+	regulator-min-microvolt = <2700000>;
+	regulator-max-microvolt = <3000000>;
+	regulator-min-dropout-voltage-microvolt = <312000>;
+};
+
 &pm8350c_flash {
 	status = "okay";
 
@@ -449,6 +510,15 @@ led-0 {
 	};
 };
 
+&pm8350c_gpios {
+	pm8008_active: pm8008-active-state {
+		pins = "gpio3";
+		function = PMIC_GPIO_FUNC_NORMAL;
+		bias-disable;
+		output-high;
+	};
+};
+
 &pmk8350_rtc {
 	status = "okay";
 };
